feat: adding lessons for else if to pyramid project (#54830)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/664599653fcd6e97104f9261.md

---
id: 664599653fcd6e97104f9261
title: Step 76
challengeType: 1
dashedName: step-76
---

# --description--

Sometimes you will want to run different code when all of the `if...else if` conditions are `false`. You can do this by adding an `else` block. 

> An `else` block will only evaluate if the `if` and `else if` blocks do not evaluate.

Reviewer: ```suggestion
An `else` block will only evaluate if the conditions in the `if` and `else if` blocks are not met.
```

} else if (condition2) {
  // this code will run if the first condition is false
} else {
  // this code will run 
  // if the first and second conditions are false
}
```

Add an `else` block to the `if` block. Inside the `else` block, log the string `"This is the else block"` to the console.

To see the results in the console, you can manually change the `<` in the `else if` statement to `>`. That will make the condition `false` and the `else` block will run.

# --hints--

You should have an `else` block.

```js
assert.match(code, /else\s*\{/);
```

Your `else` block should log the string `"This is the else block"` to the console.

```js
assert.match(code, /console\.log\(\s*["']This\s+is\s+the\s+else\s+block["']\s*\)/);
```

# --seed--

## --seed-contents--

```js
const character = "#";
const count = 8;
const rows = [];

function padRow(rowNumber, rowCount) {
  return " ".repeat(rowCount - rowNumber) + character.repeat(2 * rowNumber - 1) + " ".repeat(rowCount - rowNumber);
}

// TODO: use a different type of loop
/*for (let i = 1; i <= count; i++) {
  rows.push(padRow(i, count));
}*/

--fcc-editable-region--
if ("") {
  console.log("Condition is true");

> } else if(5 < 10) {

Reviewer 2: ```suggestion
} else if (5 < 10) {
```

  console.log("5 is less than 10");
}
--fcc-editable-region--

> let result = ""

Reviewer 2: ```suggestion
let result = "";
```
